### DISCUSSION TITLE
How to implement mqtt "message expiry interval" logic in v2.3.1

### DISCUSSION THREAD

**Discussion by OlgasAcc** [edgexfoundry/community#244] [Q&A]
Hello,
Our app-service (golang) retains memory when nodes and cores are out of the network without releasing it even when the network is back. 
QoS=0, Retain=false, PersistOnError=false.

We use edgex v2.3.1, so "transforms" library is operating within "paho.mqtt.golang" client for MQTT 3.1 protocol. AFAIK "paho.golang" has support for MQTT 5.0 and provides the option to set MessageExpiryInterval. 

Is there any other workaround to set expiration for the messages in v2.3.1?

**Comment by jiekechoo** [comment]
@OlgasAcc Thanks for your info. The current LTS version is EdgeX V3.1, maybe you could upgrade to it.

**Reply by OlgasAcc** [reply]
Hi, thanks for your answer.
I see you still use MQTT 3.0 client in v3.1.0, so Im a bit confused how could upgrade help to set messageExpiryInterval. Maybe Im missing something?
Thanks

**Comment by lenny-goodell** [comment]
@OlgasAcc , this is a new feature in MQTT 5. 
_The Message Expiry Interval is a new feature introduced in MQTT 5.0, which allows the publisher to set an expiry interval for time-sensitive messages._

EdgeX still uses the MQTT 3.1 golang client since the 5.0 version is not stable, i.e no 1.x release yet.
3.x client: https://github.com/eclipse/paho.mqtt.golang
5.x client: https://github.com/eclipse/paho.golang
